gateway-server/src/main/java/org/apache/knox/gateway/services/security/impl/DefaultAliasService.java

```diff
@@ -194,6 +195,24 @@ public void removeAliasesForCluster(String clusterName, Set<String> aliases) thr
     return getPasswordFromAliasForCluster(NO_CLUSTER_NAME, alias);
   }
 
+  //Overriding the default behavior as we want to avoid loading the keystore N-times from the file system
+  @Override
```

Comment:
Does something similar need to be done for cluster-specific aliases?

Reply:
I'm going to re-check and change other methods too if needed.

Reply:
So, in this case, the cluster we use is `__gateway` and the only reason this method was added is to avoid exhausting I/O operations in the new `AliasBasedTokenStateService.loadGatewayCredentialsOnStartup()` method.
Therefore I don't see if other methods should be changes within the scope of this task.